src/components/SIngleArticle.jsx

```diff
@@ -81,6 +85,28 @@ function SingleArticle() {
     }
   }, [isPosted]);
 
+  const deleteComment = (id) => {
+    setDeleting(true);
+    deleteCommentById(id)
+      .then((response) => {
+        setDeleting(false);
+        setIsDeleted(true);
+        setComments(comments.filter((comment) => comment.comment_id !== id))
+      })
+      .catch((error) => {
+        console.error("Error deleting comment:", error)
+        setDeleting(false)
+      });
+  };
+  useEffect(() => {
+    if (isDeleted) {
```

Comment:
if we optimistically render the posting/deleting, we shouldn't need the isDeleted, isPosting states :)